crawl-ref/source/actor.cc

```diff
@@ -516,6 +542,47 @@ void actor::stop_constricting_all(bool intentional, bool quiet)
     constricting = 0;
 }
 
+static bool _invalid_constrictee(const actor *constrictee)
+{
+    return !constrictee || !constrictee->is_constricted();
+}
+
+/**
+ * Stop directly constricting all defenders.
+ *
+ * @param intentional True if this was intentional, which affects the language
+ *                    in any message.
+ * @param quiet       If True, don't display a message.
+ */
+void actor::stop_directly_constricting_all(bool intentional, bool quiet)
```

Comment:
Good function. Your fix for my complaint about the obtuseness of `stop_constricting_all(true, false)` is better than what I had in mind.